### DISCUSSION TITLE
TTS and game development

### DISCUSSION THREAD

**Discussion by BitBarrel** [coqui-ai/TTS#1697] [General Q&A] [answered]
The new coqui.ai websites gives game design as an example use case. However, it is not possible to deploy real time TTS in games at the moment due to the lack of an API outside of a Linux environment. This issue has been raised several times before but unfortunately there is currently no interest in addressing this.

Could someone elaborate how the current state of the TTS project is suitable for game design?

**Comment by p0p4k** [comment, marked as answer]
Wouldn't you just synthesize voices beforehand and save it in the games directory for running during the gameplay? Do you want a real time TTS for every time the game is played?

**Reply by BitBarrel** [reply]
That would require an impossible amount of sound files in my use case.

**Reply by p0p4k** [reply]
Just replying to your original post, this TTS repo is totally working on my windows pc as well. I do not know your exact use case as well, so i cannot give you specific opinions too.

**Reply by BitBarrel** [reply]
I am looking for a smooth industry standard end user deployment scenario. Asking thousand of end users to install various packages, and make various configuration changes with a command line is not a 2022 experience 😏

**Reply by p0p4k** [reply]
So do you want a "smooth industry standard end user deployment scenario" to be free of cost in 2022? This repo is just a tool to get you started with TTS development. Deploying, production and neat packaging is not part of this cycle and you are supposed to figure it out. There are plenty of ways to get it done online. Next time, try to not take things for granted and be thankful for once that such fantastic high quality repositories with SOTA algorithms implemented in open source exist for free to use!

**Reply by BitBarrel** [reply]
Thank you for your response. I wish to steer away from a not-nice tone, so common online these days.

Continueing the conversation in a constructive way, why does Coqui STT have a robust deployment environment but TTS does not?

**Reply by p0p4k** [reply]
What I believe is Coqui STT just relies on one model (from deepmind, baidu) that gives really good accuracy already. It is ready for deployment in forms of production ready tflite model. With TTS, the field is ever expanding and the models are still up and coming. Converting existing models to tflite/ONNX models takes lot of work and is a part of the roadmap. 
If you try to open the [roadmap for TTS](https://github.com/coqui-ai/TTS/issues/378#:~:text=ONNX%20and%20TorchScript%20model%20exports.) production ready models are a to-do goal. Eventually it will happen, but good things take time. If you are in a hurry, you are always welcome to contribute in a constructive way. Thanks.

**Reply by BitBarrel** [reply]
Thanks, that is an explanation I was looking for. Much appreciated. 

I'm not in any hurry, so I can wait, no problem. The lack of a deployment interface was just a decision I didn't understand, but now I do.

As for contributing, I did look into that before but unfortunately my coding skills are not up to scratch and I would be into it way over my head.

**Reply by p0p4k** [reply]
No worries mate. Good day and good luck! 👍
